src/native/libs/System.Native/pal_io.c

```diff
@@ -61,12 +61,32 @@ extern int     getpeereid(int, uid_t *__restrict__, gid_t *__restrict__);
 #endif
 #endif
 
-// The portable build is performed on RHEL7 which doesn't define FICLONE yet.
-// Ensure FICLONE is defined for all Linux builds.
 #ifdef __linux__
+
+// Ensure FICLONE is defined for all Linux builds.
 #ifndef FICLONE
 #define FICLONE _IOW(0x94, 9, int)
 #endif
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wreserved-id-macro"
+// Ensure __NR_copy_file_range is defined for portable builds.
+#include <sys/syscall.h> // __NR_copy_file_range
+# if !defined(__NR_copy_file_range)
```

Comment:
Is it safe to assume these numbers on non-Linux systems? I'd expect this to be guarded somehow.

Reply:
This whole block is under `#ifdef __linux__`.

Reply:
So essentially the whole `copy_file_range` would not work on non-Linux anyway? I mean, FreeBSD supports it and the managed code intentionally allows non-Linux systems so it's a bit confusing.